x-pack/plugins/licensing/public/services/feature_usage_service.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import isDate from 'lodash/isDate';
+import type { HttpSetup, HttpStart } from 'src/core/public';
+import { LicenseType } from '../../common/types';
+
+/** @public */
+export interface FeatureUsageServiceSetup {
+  /**
+   * Register a feature to be able to notify of it's usages using the {@link FeatureUsageServiceStart | service start contract}.
+   */
+  register(featureName: string, licenseType: LicenseType): Promise<void>;
+}
+
+/** @public */
+export interface FeatureUsageServiceStart {
+  /**
+   * Notify of a registered feature usage at given time.
+   *
+   * @param featureName - the name of the feature to notify usage of
+   * @param usedAt - Either a `Date` or an unix timestamp with ms. If not specified, it will be set to the current time.
+   */
+  notifyUsage(featureName: string, usedAt?: Date | number): Promise<void>;
+}
+
+interface SetupDeps {
+  http: HttpSetup;
+}
+
+interface StartDeps {
+  http: HttpStart;
+}
+
+/**
+ * @internal
+ */
+export class FeatureUsageService {
+  public setup({ http }: SetupDeps): FeatureUsageServiceSetup {
+    return {
+      register: async (featureName, licenseType) => {
+        await http.post('/internal/licensing/feature_usage/register', {
+          body: JSON.stringify({
+            featureName,
+            licenseType,
+          }),
+        });
+      },
+    };
+  }
+
+  public start({ http }: StartDeps): FeatureUsageServiceStart {
+    return {
+      notifyUsage: async (featureName, usedAt = Date.now()) => {
+        const lastUsed = isDate(usedAt) ? usedAt.getTime() : usedAt;
+        await http.post('/internal/licensing/feature_usage/notify', {
```

Comment:
@pgayvallet, should we catch and swallow errors here? I guess consumer shouldn't care about handling errors from this (nor probably wait for a success) and unhandled promise rejection from this call would be a bummer.

Reply:
idk. Realistically, the only reason this would fails (appart from network failures) would be if the consumer tries to notify usage of an unregistered feature, which it a consumer cause (and should be detected in development).

Also, unhandled rejections on the client-side don't really do much (even if of course they should be avoided)

But that's not a strong opinion, and if really you think this is preferable, I can change that.